components/timezone/src/provider.rs

```diff
@@ -42,6 +42,7 @@ const _: () = {
     icu_timezone_data::make_provider!(Baked);
     icu_timezone_data::impl_time_zone_bcp47_to_iana_v1!(Baked);
     icu_timezone_data::impl_time_zone_iana_to_bcp47_v1!(Baked);
+    icu_timezone_data::impl_time_zone_iana_to_bcp47_v2!(Baked);
```

Comment:
if the v2 struct supports all v1 functionality, we should not include baked data for both